pyriemann/classification.py

```diff
@@ -498,19 +500,113 @@ def fit(self, X, y):
         return self
 
     def predict(self, covtest):
-        """get the predictions.
+        """Get the predictions.
 
         Parameters
         ----------
-        X : ndarray, shape (n_trials, n_channels, n_channels)
-            ndarray of SPD matrices.
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
 
         Returns
         -------
-        pred : ndarray of int, shape (n_trials, 1)
+        pred : ndarray of int, shape (n_matrices, 1)
             the prediction for each trials according to the closest centroid.
         """
         dist = self._predict_distances(covtest)
         neighbors_classes = self.classes_[np.argsort(dist)]
         out, _ = stats.mode(neighbors_classes[:, 0:self.n_neighbors], axis=1)
         return out.ravel()
+
+
+class SupportVectorMachine(BaseEstimator, ClassifierMixin):
+    """Classification by Riemannian Support Vector Machine.
+
+    Support vector machine with precomputed Riemannian kernel matrix
+    according to different metrics as described in [1]_.
+
+    Parameters
+    ----------
+    svc_func : class, (default: sklearn.svm.SVC)
+        Class that performs the svc calculations.
+    metric : {'riemann', 'euclid', 'logeuclid'}
+        Metric for kernel matrix computation.
+    Cref : None | ndarray, shape (n_channels, n_channels)
+        Reference point for kernel matrix computation. If None, the mean of
+        the training data according to the metric is used.
+    **kwargs
+        Keyword arguments passed to svc_func.
+
+    Attributes
+    ----------
+    svc_ : svc_func instance
+        Fitted SVC with precomputed Riemannian kernel matrix.
+    data_ : ndarray, shape (n_matrices, n_channels, n_channels)
+        Training data.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    References
+    ----------
+    .. [1] A. Barachant, S. Bonnet, M. Congedo, and C. Jutten.
+        Classification of covariance matrices using a Riemannian-based kernel
+        for BCI applications". In: Neurocomputing 112 (July 2013), pp. 172-178.
+    """
+
+    def __init__(self, svc_func=SVC, metric='riemann', Cref=None,
```

Comment:
SVC is not a func but a "klass" or "factory". In any case I would not expose this flexibility have write 2 new estimators (one for classif and one for regression.

Reply:
I implemented it this way to be able to use GPU accelerated SVC calculation (like thundersvm).

Reply:
Although I'm unsure if precomputed kernels can be accelerated much by GPU.

Reply:
I'll ask for a confirmation, but from what I remember, there no specific gain with GPU for precomputed kernels.